common/src/main/java/com/wynntils/models/items/encoding/impl/block/NameDataTransformer.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © Wynntils 2023-2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.items.encoding.impl.block;
+
+import com.wynntils.models.items.encoding.data.NameData;
+import com.wynntils.models.items.encoding.type.DataTransformer;
+import com.wynntils.models.items.encoding.type.DataTransformerType;
+import com.wynntils.models.items.encoding.type.ItemTransformingVersion;
+import com.wynntils.utils.UnsignedByteUtils;
+import com.wynntils.utils.type.ArrayReader;
+import com.wynntils.utils.type.ErrorOr;
+import com.wynntils.utils.type.UnsignedByte;
+import java.util.ArrayList;
+import java.util.List;
+
+public class NameDataTransformer extends DataTransformer<NameData> {
+    @Override
+    public ErrorOr<UnsignedByte[]> encodeData(ItemTransformingVersion version, NameData data) {
+        return switch (version) {
+            case VERSION_1 -> encodeName(data.name());
+        };
+    }
+
+    @Override
+    public ErrorOr<NameData> decodeData(ItemTransformingVersion version, ArrayReader<UnsignedByte> byteReader) {
+        return switch (version) {
+            case VERSION_1 -> decodeName(byteReader);
+        };
+    }
+
+    private ErrorOr<UnsignedByte[]> encodeName(String name) {
+        try {
+            return ErrorOr.of(UnsignedByteUtils.encodeString(name));
```

Comment:
Wasn't the point that the name should not be byte-encoded, so at least it will show for non-Wynntils users? Or maybe I'm missing something in the grand scheme of things how the chat encoding packet is sent...

Reply:
It was the point yes. But it turned out to be really messy, so I opted to just encode the name. Hopefully it is not a big issue, and if it is, we can fix it up with a new version.

Reply:
Perhaps if we put the name as part of the start or end packet it gets simpler? I mean, if we want a non-encoded string it is more part of the packet itself, not the individual data blocks. So maybe we could sneak it in in the start block, and when we encode the start block, we extract the name and send it separately..?

Reply:
If we want to do something like that, it is probably best to do it now, since that changes the entire packet concept, rather than an individual data packet. Or, hm, maybe we should do both. That is, both keep this for the actual name, and then add some kind of "prefix" to the packet containing a "user representation", like `<start user representation byte>User visible name for non-wynntils users<the entire encoded packet>`.

Reply:
Let me share my thinking. Normally, I would say, yes, let's do this, give everyone the best experience. But this case, it's really tricky to get the code quality/system right, without breaking major abstractions. Vanilla users feel left out, yes. However, they are basically left out either way, Wynncraft items tend to be special not because of rarity, but their stats. So knowing that "I am selling x mythic" is not any better than knowing "I am selling x" without knowing it's stats.

The whole `EncodedByteBuffer` concept relies on the fact that everything is encoded. This (sadly) includes names too. We even "gain" a little, we can encode 2 ASCII chars into a single Unicode character.

If this becomes a problem for Wynncraft, I'll personally work on fixing it. However, I think we should keep it as is for now.